src/main/java/si/mazi/rescu/clients/ApacheConnection.java

```diff
@@ -0,0 +1,228 @@
+package si.mazi.rescu.clients;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetSocketAddress;
+import java.net.MalformedURLException;
+import java.net.ProtocolException;
+import java.net.Proxy;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLSocketFactory;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpHost;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpDelete;
+import org.apache.http.client.methods.HttpEntityEnclosingRequestBase;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpHead;
+import org.apache.http.client.methods.HttpOptions;
+import org.apache.http.client.methods.HttpPatch;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.entity.ByteArrayEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import si.mazi.rescu.HttpMethod;
+
+public class ApacheConnection implements HttpConnection {
+    
+    private static final Logger log = LoggerFactory.getLogger(ApacheConnection.class);
+    
+    private final String url;
+    private final Proxy proxy;
+    private final HttpClientBuilder builder = HttpClients.custom();
+    private final RequestConfig.Builder requestConfig = RequestConfig.custom();
+            
+    private boolean executed = false;
+    private CloseableHttpResponse res;
+    private HttpMethod method = HttpMethod.GET;
+    private ByteArrayOutputStream out;
+    private final Map<String, String> headers = new HashMap<>();
+    
+    private CloseableHttpResponse response;
+
+    private ApacheConnection(String url, Proxy proxy) {
+        super();
+        this.url = url;
+        this.proxy = proxy;
+    }
+    
+    public static HttpConnection create(String url, Proxy proxy) throws MalformedURLException, IOException {
+        return new ApacheConnection(url, proxy);
+    }
+    
+    private void exec() {
```

Comment:
Should declare throws IOException (and not wrap in RuntimeException) and possibly other useful exceptions